### ISSUE TITLE
clean up schema-diff

### ISSUE THREAD

**Issue by whit537** [gratipay/gratipay.com#3471]
By making the payday tables permanent in #3441, we introduced schema drift. `make schema-diff` now fails.
##

**Comment by whit537** [comment]
+1 from @rohitpaulk in https://github.com/gratipay/gratipay.com/issues/3471:

> `make schema-diff` errors out when deploying code. It complains about differences between the local and production DB, and all of them are related to payday. This happens because we don't delete all payday related tables and functions in our production DB until the next payday starts. This change was intentional - it helps us debug issues related to payday. I can't find the exact PR where the change was made, if someone does please link it here.
>
One solution that I see here is to run `payday.sql` against the test DB before generating `local.sql`.

**Comment by whit537** [comment]
Me over there:

> We could also make the `payday_` tables permanent instead of temporary.

**Comment by whit537** [comment]
This change was insufficient:

```diff
diff --git a/recreate-schema.sh b/recreate-schema.sh
index 2492343..0846bd8 100755
--- a/recreate-schema.sh
+++ b/recreate-schema.sh
@@ -24,6 +24,7 @@ echo "Applying sql/schema.sql ..."
 echo 
 
 psql $DATABASE_URL < sql/schema.sql
+psql $DATABASE_URL < sql/payday.sql
 
 
 echo "=============================================================================="
```

**Comment by rohitpaulk** [comment]
I'm seeing a few differences - one is that the production database has a table called 'affected'. Looks like this was used for sending a temporary email or so? Checking issues..

**Comment by rohitpaulk** [comment]
Hmm, doesn't look like it was under https://github.com/gratipay/gratipay.com/issues/4393

**Comment by rohitpaulk** [comment]
I don't see any code anywhere that uses it, going to delete.

**Comment by rohitpaulk** [comment]
Deleted.

**Comment by rohitpaulk** [comment]
One more: `payday_subscriptions` is present on production, but not local. I don't see code that uses it either, maybe part of the old payday script?

**Comment by rohitpaulk** [comment]
Yep, removed in https://github.com/gratipay/gratipay.com/commit/151bda0a2855715d46ab246fbad35b57b35b1b49. 

I'm deleting the table on production

**Comment by rohitpaulk** [comment]
![schema](https://cloud.githubusercontent.com/assets/3893573/25771817/f11645d2-3279-11e7-8664-370356ec75cf.png)

One old function lying around from when we were using usernames instead of IDs. I didn't know that Postgres supported functions with the same name but different signatures...

**Comment by rohitpaulk** [comment]
Down to a single statement now: 

![screen shot 2017-05-07 at 2 44 48 am](https://cloud.githubusercontent.com/assets/3893573/25775961/345804fc-32cf-11e7-9136-0f713d023f12.png)

**Comment by rohitpaulk** [comment]
^ Fixed that by adding `--no-privileges` when calling `pg_dump`

**Comment by whit537** [comment]
Closed in #4450.

**Comment by whit537** [comment]
Closed in #4450.
